comp/core/hostname/hostnameinterface/interface.go

```diff
@@ -0,0 +1,27 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package hostnameinterface describes the interface for hostname methods
+package hostnameinterface
+
+import (
+	"context"
+)
+
+// Data contains hostname and the hostname provider
+type Data struct {
+	Hostname string
+	Provider string
+}
+
+// HostnameInterface is the type for hostname methods.
+type HostnameInterface interface {
```

Comment:
`hostnameinterface.Hostnameinterface` is odd to use. Please rename this to component or something else .